programs/create-genesis-ee/genesis_ee_builder.cpp

```diff
@@ -0,0 +1,186 @@
+    maps_.add_index<comment_header_index>();
+}
+
+genesis_ee_builder::~genesis_ee_builder() {
+}
+
+golos_dump_header genesis_ee_builder::read_header(bfs::fstream& in, uint32_t expected_version) {
+    golos_dump_header h{"", 0, 0};
+    in.read((char*)&h, sizeof(h));
+    if (in) {
+        EOS_ASSERT(std::string(h.magic) == golos_dump_header::expected_magic && h.version == expected_version, genesis_exception,
+            "Unknown format of the operation dump file.");
+    }
+    return h;
+}
+
+bool genesis_ee_builder::is_newer(const golos_dump_header& h, const comment_header& comment) {
+    return h.block_num > comment.block_num || (h.block_num == comment.block_num && h.op_in_block > comment.op_in_block);
+}
+
+void genesis_ee_builder::process_comments() {
+    std::cout << "-> Reading comments..." << std::endl;
+
+    bfs::fstream in(in_dump_dir_ / "comments");
+    while (in) {
+        auto h = read_header(in, 1);
+
+        if (!in) {
+            break;
+        }
+
+        uint64_t hash = 0;
+        fc::raw::unpack(in, hash);
+
+        auto comment_offset = uint64_t(in.tellg());
+
+        cyberway::golos::comment_operation cop;
+        fc::raw::unpack(in, cop);
+
+        const auto& comment_index = maps_.get_index<comment_header_index, by_hash>();
+        auto comment_itr = comment_index.find(hash);
+        if (comment_itr != comment_index.end()) {
+            maps_.remove(*comment_itr);
+        }
+
+        maps_.create<comment_header>([&](auto& comment) {
+            comment.hash = hash;
+            comment.offset = comment_offset;
+            comment.block_num = h.block_num;
+            comment.op_in_block = h.op_in_block;
+        });
+    }
+}
+
+void genesis_ee_builder::process_delete_comments() {
+    std::cout << "-> Reading comment deletions..." << std::endl;
+
+    bfs::fstream in(in_dump_dir_ / "delete_comments");
+    while (in) {
+        auto h = read_header(in, 1);
+
+        if (!in) {
+            break;
+        }
+
+        uint64_t hash = 0;
+        fc::raw::unpack(in, hash);
+
+        const auto& comment_index = maps_.get_index<comment_header_index, by_hash>();
+        auto comment_itr = comment_index.find(hash);
+        if (is_newer(h, *comment_itr)) {
+            maps_.remove(*comment_itr);
+        }
+    }
+}
+
+void genesis_ee_builder::process_rewards() {
+    std::cout << "-> Reading rewards..." << std::endl;
+
+    bfs::fstream in(in_dump_dir_ / "total_comment_rewards");
+    while (in) {
+        auto h = read_header(in, 1);
+
+        if (!in) {
+            break;
+        }
+
+        uint64_t hash = 0;
+        fc::raw::unpack(in, hash);
+
+        cyberway::golos::total_comment_reward_operation tcrop;
+        fc::raw::unpack(in, tcrop);
+
+        int64_t net_rshares = 0;
+        fc::raw::unpack(in, net_rshares);
+
+        const auto& comment_index = maps_.get_index<comment_header_index, by_hash>();
+        auto comment_itr = comment_index.find(hash);
```

Comment:
You should implement a function to check that hash equal to `author-permlink`
We should fix hash-function, it will be critical if posts got rewards from other post.

Reply:
If it so, we can use a bigger type - uint128 and uint256 - for more distribution of hash values.